src/entity/Ship.java

```diff
@@ -56,37 +59,86 @@ public Ship(final int positionX, final int positionY, final Direction direction)
     }
 
     /**
-     * Moves the ship speed uni ts right, or until the right screen border is reached.
+     * Moves the ship right until the right screen border is reached.
      */
     public final void moveRight() {
         this.direction = Direction.RIGHT;
         this.positionX += speed;
     }
 
     /**
-     * Moves the ship speed units left, or until the left screen border is reached.
+     * Moves the ship left until the left screen border is reached.
      */
     public final void moveLeft() {
         this.direction = Direction.LEFT;
         this.positionX -= speed;
     }
 
     /**
-     * Moves the ship speed units up, or until the top screen border is reached.
+     * Moves the ship up until the top screen border is reached.
      */
     public final void moveUp() {
         this.direction = Direction.UP;
         this.positionY -= speed;
     }
 
     /**
-     * Moves the ship speed units down, or until the bottom screen border is reached.
+     * Moves the ship down until the bottom screen border is reached.
      */
     public final void moveDown() {
         this.direction = Direction.DOWN;
         this.positionY += speed;
     }
 
+    /**
+     * Moves the ship up the right until the top and right screen border is reached.
+     */
+    public final void moveUpRight() {
+        this.direction = Direction.UP_RIGHT;
+        calculateMovement();
+        this.positionY -= movement;
+        this.positionX += movement;
+    }
+
+    /**
+     * Moves the ship up the left until the top and left screen border is reached.
+     */
+    public final void moveUpLeft() {
+        this.direction = Direction.UP_LEFT;
+        calculateMovement();
+        this.positionY -= movement;
+        this.positionX -= movement;
+    }
+
+    /**
+     * Moves the ship down the right until the bottom and right screen border is reached.
+     */
+    public final void moveDownRight() {
+        this.direction = Direction.DOWN_RIGHT;
+        calculateMovement();
+        this.positionY += movement;
+        this.positionX += movement;
+    }
+
+    /**
+     * Moves the ship down the left until the bottom and left screen border is reached.
+     */
+    public final void moveDownLeft() {
+        this.direction = Direction.DOWN_LEFT;
+        calculateMovement();
+        this.positionY += movement;
+        this.positionX -= movement;
+    }
+
+    /**
+     * 축 방향 이동속도에서 소수점 아래 부분 누적 및 정수 부분 구분.
+     */
+    private void calculateMovement() {
+        remainingMovement += (int) (speed / Math.sqrt(2));
```

Comment:
대각선 방향 이동이 기본 이동보다 더 느린 것 같아 확인해보니 calculateMovement() 메소드에 문제가 있네요.

remainingMovement에 처음 값을 대입할 때 int로 type casting으로 인해 소수점이 사라지는 문제가 있습니다.

예를 들어 speed = 2라면, speed / Math.sqrt(2) = 1.414... 인데 int로 type casting 되어서 1이 됩니다.
따라서 remainingMovement는 그냥 1이 되고 이후에 movement도 1, 그리고 remainingMovement는 계속 0이 됩니다.

다음과 같이 int type casting을 제거해야합니다.
```suggestion
        remainingMovement += speed / Math.sqrt(2);
```

Reply:
확인하고 수정했습니다!